psl/parser-database/src/types.rs

```diff
@@ -263,6 +276,42 @@ impl ScalarFieldType {
     pub fn is_decimal(self) -> bool {
         matches!(self, Self::BuiltInScalar(ScalarType::Decimal))
     }
+
+    /// Display the field type as it would appear in the Prisma schema.
+    pub fn display<'a>(&'a self, db: &'a ParserDatabase) -> impl fmt::Display + 'a {
+        DisplayScalarFieldType { field_type: self, db }
+    }
+}
+
+struct DisplayScalarFieldType<'a> {
+    pub field_type: &'a ScalarFieldType,
+    pub db: &'a ParserDatabase,
+}
+
+impl fmt::Display for DisplayScalarFieldType<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.field_type {
+            ScalarFieldType::BuiltInScalar(t) => write!(f, "{}", t.as_str()),
```

Comment:
I think it would actually make sense for `ScalarType` to implement `Display` but I'm fine with it either way